Wk/types 001 - Vouchers (ListGiftCardTransactions, ExportGiftCardTransactions)

weronika-kurczyna opened the pull request:
**New exported types/interfaces**
- ListGiftCardTransactionsResponse
- ListGiftCardTransactionsType
- ListGiftCardTransactionsTypeDataObject
- ExportGiftCardTransactionsBody
- ExportGiftCardTransactionsResponse
- ExportGiftCardTransactionsParams
- ExportGiftCardTransactionsFields
- ExportGiftCardTransactionsParamsFilters

Added support for following endpoints:

- vouchers
    - GET /vouchers/{code}/transactions
    - POST /vouchers/{code}/transactions/export
    
Note: ListGiftCardTransactions response includes also `total` property (which is only visible when there are not transactions). This property was previously omitted in openApi file.

darekg11 committed:
[patch] Add option to pass timeout to VoucherifyServerSide client and VoucherifyClientSide client (#223)

* patch - add optional timeoutMs option to VoucherifyServerSide and VoucherifyClientSide classes defiing timeout in miliseconds after which Axios is going to abort the request. By default this is equal to 0 meaining that there is no timeout beside default Voucherify's Load balancer timeout which is set to 3 minutes

* Add CHANGESET file

* Fix indentation

darekg11 committed:
Auto updated documentation

github-actions[bot] committed:
Version Packages (#224)

Co-authored-by: github-actions[bot] <github-actions[bot]@users.noreply.github.com>

darekg11 committed:
Auto updated documentation

weronika-kurczyna committed:
add listGiftCardTransactions method

weronika-kurczyna committed:
Add tests for List Gift Card Transactions

weronika-kurczyna committed:
export gift card transactions

weronika-kurczyna committed:
add tests for List Gift Card Transactions Export

weronika-kurczyna committed:
add enum to source in DataObject

weronika-kurczyna committed:
Update Vouchers.ts

weronika-kurczyna committed:
Update new-endpoints-vouchers.md

weronika-kurczyna committed:
Update new-endpoints-vouchers.md

p-zielinski commented on packages/sdk/src/types/Vouchers.ts:
```diff
@@ -251,3 +251,112 @@ export type VouchersBulkUpdateMetadataResponse = {
+
+export interface ExportGiftCardTransactionsBody {
+	parameters?: {
+		order?: '-created_at' | 'created_at'
```
this shall be equal to ExportGiftCardTransactionsFields and ExportGiftCardTransactionsFields.map(e => `-${e}`)

weronika-kurczyna replied:
do you mean that each field in ExportGiftCardTransactionsFields can be an `order` value (descending or ascending)?

weronika-kurczyna replied:
even if you call the export from the V% dashboard the only possible options are '-created_at' | 'created_at'

p-zielinski commented on packages/sdk/src/types/Vouchers.ts:
```diff
@@ -251,3 +251,112 @@ export type VouchersBulkUpdateMetadataResponse = {
+export interface ListGiftCardTransactionsResponse {
+	object: 'list'
+	data_ref: 'data'
+	data: ListGiftCardTransactionsTypeDataObject[] | []
```
simply `ListGiftCardTransactionsTypeDataObject[]` ?

p-zielinski commented on packages/sdk/src/types/Vouchers.ts:
```diff
@@ -251,3 +251,112 @@ export type VouchersBulkUpdateMetadataResponse = {
+	source_id: string | null
+	voucher_id: string
+	campaign_id: string | null
+	source: 'voucherify-web-ui' | 'API' | null
```
are you sure about that?

weronika-kurczyna replied:
I received a response in which the `source` is `null`
The `id` is always in the response (id of the transaction)
`source_id` is not defined in many cases, but it's returned in every response
`voucher_id` must be defined
if it's a standalone the `campaign_id` could be null

weronika-kurczyna replied:
or perhaps you are referring to some specific property? :)

p-zielinski commented on packages/sdk/src/Vouchers.ts:
```diff
@@ -89,4 +89,16 @@ export class Vouchers {
+	 * @see https://docs.voucherify.io/reference/export-gift-card-transactions
+	 */
+	public exportGiftCardTransactions(code: string, body: T.ExportGiftCardTransactionsBody) {
+		return this.client.post<T.ExportGiftCardTransactionsResponse>(`/vouchers/${encode(code)}/transactions/export`, body)
```
just to be sure?
Is there a case when you will get
```
{
...
status: DONE
result: string
}
```
here?

weronika-kurczyna replied:
it looks like it's an endpoint dedicated only to schedule the export and it always returns "status": "SCHEDULED", need to check this

weronika-kurczyna committed:
add loyalty card to type enum

weronika-kurczyna committed:
Update Vouchers.ts

weronika-kurczyna committed:
Update Vouchers.ts

weronika-kurczyna commented:
@p-zielinski I hope the changes will be correct, I tried by best, If you have any comments, I will change it ;)

p-zielinski committed:
Frame for integration tests + update dependencies

- Update axios from 0.21.4 to 0.27.2 in the SDK package
- Added dotenv 16.3.1 to SDK package dev dependencies.
- Added frame for integration tests

marcin-slezak committed:
Auto updated documentation

marcin-slezak committed:
Pz/types 002 - Added support for missing CSV import endpoints (#227)

- Added support for missing CSV import endpoints:
  - `POST /skus/importCSV`
  - `POST /products/importCSV`
  - `POST /vouchers/importCSV`
  - `POST /customers/importCSV`
  - `POST /campaigns/campaignId/importCSV` 
- Added form-data `4.0.0` to SDK package

marcin-slezak committed:
Auto updated documentation

marcin-slezak committed:
Pz/types 003 - Add missing methods to Categories and Campaigns API (#228)

Add missing methods covering Categories and Campaigns API.
  - Added support for new endpoints [(examples of usage available in readme.md)](..%2F..%2Fpackages%2Fsdk%2FREADME.md): 
    - `POST /campaigns/campaignId/enable`,
    - `POST /campaigns/campaignId/disable`,
    - `GET, POST DELETE /categories`,
    - `GET PUT /categories/id`.
  - New exported types/interfaces in `Categories.ts`:
    - `ListCategories`,
    - `CategoryObject`,
    - `CreateCategory`,
    - `ResponseCreateCategory`,
    - `ResponseUpdateCategory`,
    - `UpdateCategoryRequest`.

marcin-slezak committed:
Auto updated documentation

github-actions[bot] committed:
Version Packages (#226)

Co-authored-by: github-actions[bot] <github-actions[bot]@users.noreply.github.com>

marcin-slezak committed:
Auto updated documentation

weronika-kurczyna committed:
fix naming

weronika-kurczyna committed:
create changeset

weronika-kurczyna committed:
Merge branch 'main' into wk/types-001

weronika-kurczyna committed:
add type for each transaction type

weronika-kurczyna committed:
Update Vouchers.ts

marcin-slezak committed:
Pz/types 004 - Add missing Customers API endpoints (#229)

* changes to validateVoucher

* simple validations tests

* changelog

* Update tidy-comics-hear.md

* node 16.20

* node 16

* requested changes

* Update tidy-comics-hear.md

* done

* Update README.md

* Create customers.spec.ts

* Update Customers.ts

* Create tidy-ironman-two.md

* Update Customers.ts

* Update generateRandomString.ts

* requested changes

* Update package.json

* Update package.json

* Update tidy-comics-hear.md

* Revert "node 16.20"

This reverts commit f869c7992c441da3123fb536ff93e7a916ff2cd0.

* minor changes

* Delete createCampaignWithOnePromotionTierAndRandomName.ts

* requested changes

* docs

* Update tidy-ironman-two.md

* Revert "docs"

This reverts commit 4d1aa28f347bb0ac86634439c85c36fb3538ef87.

* remove validations

* changeset

* Update tidy-comics-hear.md

* Update tidy-comics-hear.md

* Update README.md

* Update tidy-comics-hear.md

* Update tidy-comics-hear.md

* Revert "Update tidy-comics-hear.md"

This reverts commit 61bbc5a7beccced47e6206e69174f901c4c232d7.

* Revert "Update tidy-comics-hear.md"

This reverts commit e6fd9766b531c421c748bf2e2cb055ca838d3c77.

* Update tidy-ironman-two.md

* Update tidy-comics-hear.md

* Create CHANGESET-TAMPLATE.md

* Update tidy-comics-hear.md

* Update CHANGESET-TAMPLATE.md

* Update CHANGESET-TAMPLATE.md

* Update CHANGESET-TAMPLATE.md

* Update CHANGESET-TAMPLATE.md

* Update tidy-comics-hear.md

* Update CHANGESET-TAMPLATE.md

* Update tidy-ironman-two.md

* Update tidy-ironman-two.md

* Update CHANGESET-TAMPLATE.md

* Update CHANGESET-TAMPLATE.md

* Update tidy-ironman-two.md

* dotenv in packages/sdk package.json

* requested changes

* requested changes

* requested changes

* Update tidy-comics-hear.md

* Delete tidy-comics-hear.md

* done

* Update ValidationsValidateVoucherResponse.html

* Update package.json

* update naming

* CustomersDeletePermanently

* typo

* typo

* revert permanent_deletion to pernament_deletion :)

* add birthdate to request

* guildeline

* delete `customers` from method

* add Customer building blocks

* align with original code from sdk (legacy)

* Update Customers.ts

* update changeset

* Update CONTRIBUTING.md

* fix description of domain types and CustomersUpdateInBulkRequestBody

* Update customers.spec.ts

* delete almost all properties from data_json (they could be different each time)

* add nullable to customer

* any -> uknown

---------

Co-authored-by: p-zielinski <82354717+p-zielinski@users.noreply.github.com>
Co-authored-by: weronika-kurczyna <117282008+weronika-kurczyna@users.noreply.github.com>
Co-authored-by: Marcin Slezak <marcin.slezak@voucherify.io>

marcin-slezak committed:
Auto updated documentation

p-zielinski committed:
Pz/types 007 - EarningRules (#241)

* done

* Update tidy-comics-tall.md

* done

* Update tidy-comics-tall.md

* test

* Update Loyalties.ts

* naming

* Update tidy-comics-tall.md

* Update Loyalties.ts

* Update tidy-comics-tall.md

* Update Loyalties.ts

marcin-slezak committed:
Auto updated documentation

github-actions[bot] committed:
Version Packages (#243)

* Version Packages

* Update CHANGELOG.md

fix naming

---------

Co-authored-by: github-actions[bot] <github-actions[bot]@users.noreply.github.com>
Co-authored-by: Marcin Ślęzak <35666945+marcin-slezak@users.noreply.github.com>

marcin-slezak committed:
Auto updated documentation

p-zielinski committed:
Pz/types 010 - loyalties members (#238)

* changes to validateVoucher

* simple validations tests

* changelog

* Update tidy-comics-hear.md

* node 16.20

* node 16

* requested changes

* Update tidy-comics-hear.md

* Update generateRandomString.ts

* requested changes

* Update package.json

* Update package.json

* Update tidy-comics-hear.md

* Revert "node 16.20"

This reverts commit f869c7992c441da3123fb536ff93e7a916ff2cd0.

* minor changes

* Delete createCampaignWithOnePromotionTierAndRandomName.ts

* requested changes

* docs

* Revert "docs"

This reverts commit 4d1aa28f347bb0ac86634439c85c36fb3538ef87.

* remove validations

* changeset

* Update tidy-comics-hear.md

* Update tidy-comics-hear.md

* Update README.md

* Update tidy-comics-hear.md

* Create CHANGESET-TAMPLATE.md

* Update tidy-comics-hear.md

* Update CHANGESET-TAMPLATE.md

* Update CHANGESET-TAMPLATE.md

* Update CHANGESET-TAMPLATE.md

* Update Loyalties.ts

* in progress

* Update CHANGESET-TAMPLATE.md

* Update tidy-comics-hear.md

* in progress

* Update CHANGESET-TAMPLATE.md

* Update CHANGESET-TAMPLATE.md

* Update CHANGESET-TAMPLATE.md

* in progress

* readme, types

* Update tidy-comics-five.md

* Update Loyalties.ts

* fixes

* minor fixes

* dotenv in packages/sdk package.json

* requested changes

* Update tidy-comics-hear.md

* Delete tidy-comics-hear.md

* tests

* Update loyalties.spec.ts

* not finished, few changes

* methods, readme

* in progress

* naming

* Update tidy-comics-five.md

* Update README.md

* Update README.md

* new types

* export and changeset

* Update Loyalties.ts

* Update Loyalties.ts

* fix

* Update Loyalties.ts

* Update Loyalties.ts

* fix

marcin-slezak committed:
Auto updated documentation

p-zielinski committed:
Pz/types 005 - Exports (#244)

* in progress

* Update Exports.ts

* done?

* Update tidy-king-sun.md

* Create exports.spec.ts

* Update exports.spec.ts

* Update exports.spec.ts

* Update Exports.ts

marcin-slezak committed:
Auto updated documentation

p-zielinski committed:
Pz/types 011 - Loyalties (includes changes from 010) (#246)

* changes to validateVoucher

* simple validations tests

* changelog

* Update tidy-comics-hear.md

* node 16.20

* node 16

* requested changes

* Update tidy-comics-hear.md

* Update generateRandomString.ts

* requested changes

* Update package.json

* Update package.json

* Update tidy-comics-hear.md

* Revert "node 16.20"

This reverts commit f869c7992c441da3123fb536ff93e7a916ff2cd0.

* minor changes

* Delete createCampaignWithOnePromotionTierAndRandomName.ts

* requested changes

* docs

* Revert "docs"

This reverts commit 4d1aa28f347bb0ac86634439c85c36fb3538ef87.

* remove validations

* changeset

* Update tidy-comics-hear.md

* Update tidy-comics-hear.md

* Update README.md

* Update tidy-comics-hear.md

* Create CHANGESET-TAMPLATE.md

* Update tidy-comics-hear.md

* Update CHANGESET-TAMPLATE.md

* Update CHANGESET-TAMPLATE.md

* Update CHANGESET-TAMPLATE.md

* Update Loyalties.ts

* in progress

* Update CHANGESET-TAMPLATE.md

* Update tidy-comics-hear.md

* in progress

* Update CHANGESET-TAMPLATE.md

* Update CHANGESET-TAMPLATE.md

* Update CHANGESET-TAMPLATE.md

* in progress

* readme, types

* Update tidy-comics-five.md

* Update Loyalties.ts

* fixes

* minor fixes

* dotenv in packages/sdk package.json

* requested changes

* Update tidy-comics-hear.md

* Delete tidy-comics-hear.md

* tests

* Update loyalties.spec.ts

* not finished, few changes

* methods, readme

* in progress

* naming

* Update tidy-comics-five.md

* Update README.md

* Update README.md

* new types

* export and changeset

* Update Loyalties.ts

* Update Loyalties.ts

* fix

* Update Loyalties.ts

* Update Loyalties.ts

* fix

* new endpoints, and change log

* readme

* new endpoint

* name of method and readme

* in progress

* Revert "in progress"

This reverts commit 89f945f5acdb8c420050ed6f261c2a604fe35ca3.

* Update Loyalties.ts

* Update Loyalties.ts

marcin-slezak committed:
Auto updated documentation

weronika-kurczyna committed:
009 - New product collections methods (#242)

* changes to validateVoucher

* simple validations tests

* changelog

* Update tidy-comics-hear.md

* node 16.20

* node 16

* requested changes

* Update tidy-comics-hear.md

* Update generateRandomString.ts

* requested changes

* Update package.json

* Update package.json

* Update tidy-comics-hear.md

* Revert "node 16.20"

This reverts commit f869c7992c441da3123fb536ff93e7a916ff2cd0.

* minor changes

* Delete createCampaignWithOnePromotionTierAndRandomName.ts

* requested changes

* docs

* Revert "docs"

This reverts commit 4d1aa28f347bb0ac86634439c85c36fb3538ef87.

* remove validations

* changeset

* Update tidy-comics-hear.md

* Update tidy-comics-hear.md

* Update README.md

* Update tidy-comics-hear.md

* Create CHANGESET-TAMPLATE.md

* Update tidy-comics-hear.md

* Update CHANGESET-TAMPLATE.md

* Update CHANGESET-TAMPLATE.md

* Update CHANGESET-TAMPLATE.md

* in progress

* done

* done

* Update ProductCollections.ts

* Update CHANGESET-TAMPLATE.md

* Update tidy-comics-hear.md

* Update CHANGESET-TAMPLATE.md

* Update CHANGESET-TAMPLATE.md

* Update CHANGESET-TAMPLATE.md

* Update tidy-comics-new.md

* dotenv in packages/sdk package.json

* requested changes

* Update tidy-comics-new.md

* few new tests, in progress

* Update ProductCollections.ts

* Update tidy-comics-hear.md

* Update package.json

* Delete tidy-comics-hear.md

* resolving conflicts

* :(

* Revert ":("

This reverts commit 1d4e35bd29abbcb74067c43b7c7e768c745dae64.

* Update ProductCollections.ts

* domain and 0-level types

* use domain and 0-level types

* fix test

* change ProductCollectionResponseData to ProductCollectionSaved

* move utility types to separate file, update changeset

* update changeset

* fix export of utility types

* fix missing object property for ProductIdentity and SkuIdentity

* update readme

* change any to unknown

* restore missing definitions from readme

* add type assertion to tests

* get rid of types that are used only in one place

* update changeset

* add missing filters keys, do not use required for domain types

---------

Co-authored-by: p-zielinski <82354717+p-zielinski@users.noreply.github.com>

marcin-slezak committed:
Auto updated documentation

github-actions[bot] committed:
Version Packages (#247)

Co-authored-by: github-actions[bot] <github-actions[bot]@users.noreply.github.com>

marcin-slezak committed:
Auto updated documentation

weronika-kurczyna committed:
Merge branch 'voucherifyio:main' into wk/types-001

weronika-kurczyna committed:
list transactions - fix order and naming of types

weronika-kurczyna committed:
export voucher transactions

weronika-kurczyna committed:
change structure to domain and 0-level types

weronika-kurczyna committed:
Update bright-mirrors-clean.md
